File: app/src/main/java/com/hjc/reader/ui/menu/adapter/LinkAdapter.java
package com.hjc.reader.ui.menu.adapter;


import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.viewholder.BaseViewHolder;
import com.hjc.reader.R;
import com.hjc.reader.bean.response.CollectLinkBean;
import com.hjc.reader.databinding.ItemLinkBinding;


public class LinkAdapter extends BaseQuickAdapter<CollectLinkBean.DataBean, BaseViewHolder> {

    public LinkAdapter() {
        super(R.layout.item_link) ;
    }

    @Override
    protected void onItemViewHolderCreated(@NonNull BaseViewHolder viewHolder, int viewType) {
        DataBindingUtil.bind(viewHolder.itemView);
    }

    @Override
    protected void convert(@NonNull BaseViewHolder helper, CollectLinkBean.DataBean item) {
        if (item == null) {
            return;
        }

        ItemLinkBinding binding = helper.getBinding();
        if (binding != null) {
            binding.setLinkBean(item);
        }
    }
}

File: module-main/src/main/java/com/hjc/learn/main/adapter/LinkAdapter.kt
package com.hjc.learn.main.adapter

import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.viewholder.BaseViewHolder
import com.hjc.learn.main.R
import com.hjc.learn.main.databinding.MainItemLinkBinding
import com.hjc.library_net.model.WanCollectLinkBean

/**
 * @Author: HJC
 * @Date: 2021/3/3 10:35
 * @Description: 收藏网址adapter
 */
class LinkAdapter(data: MutableList<WanCollectLinkBean>?) :
    BaseQuickAdapter<WanCollectLinkBean, BaseViewHolder>(R.layout.main_item_link, data) {

    override fun onItemViewHolderCreated(viewHolder: BaseViewHolder, viewType: Int) {
        DataBindingUtil.bind<ViewDataBinding>(viewHolder.itemView)
    }

    override fun convert(holder: BaseViewHolder, item: WanCollectLinkBean) {
        val binding = DataBindingUtil.getBinding<MainItemLinkBinding>(holder.itemView)
        binding?.linkBean = item
    }
}